### PR TITLE
🧵 Add support for `UFFD_FEATURE_THREAD_ID`

### PR DESCRIPTION
**Pull request by acfoltzer**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "userfaultfd"
-version = "0.3.0"
+version = "0.3.1"
```
Comment (acfoltzer) [inline Cargo.toml:3]: Minor version bump because the would-be breaking change is behind a new cargo feature

File: src/event.rs [hunk 2]
```diff
@@ -19,6 +21,15 @@ pub enum Event {
         rw: ReadWrite,
         /// The address that triggered the fault.
         addr: *mut c_void,
+        /// The thread that triggered the fault, if [`FeatureFlags::THREAD_ID`] is enabled.
+        ///
+        /// If the thread ID feature is not enabled, the value of this field is undefined. It would
+        /// not be undefined behavior to use it, strictly speaking, but the [`Pid`] will not
+        /// necessarily point to a real thread.
```
Comment (acfoltzer) [inline src/event.rs:28]: This, uh, sucks. But there's no way to look at the event in isolation and know whether this is a valid pid or not

**Review by pchickey — APPROVED** [review]

File: src/event.rs (hunk 2 above)
Comment (awortman-fastly) [reply src/event.rs:28]: me: "what if we wrap this fd with some extra data to indicate later on if the pid should be valid, ourselves?"

Adam: turns out all we get during a fork event is the userfault fd and _not_ flags, so this is inconsistent with using and supporting uffd across fork events.

this really is the best option for the time being :grimacing:

**Review by awortman-fastly — APPROVED** [review]
ughhhhhhhhhhh at all of this. but thank you for giving this userfaultfd-rs bindings!

**Comment by acfoltzer** [comment]
Released as 0.3.1 :tada:

**Pull request merged**